src/antq/util/git.clj

```diff
@@ -1,19 +1,39 @@
 (ns antq.util.git
   (:require
+   [antq.log :as log]
    [clojure.java.shell :as sh]
    [clojure.string :as str]))
 
 (defn- extract-tags
   [ls-remote-resp]
-  (->> (:out ls-remote-resp)
+  (some->> (:out ls-remote-resp)
        (str/split-lines)
        (keep #(second (str/split % #"\t" 2)))
        (filter #(= 0 (.indexOf ^String % "refs/tags")))
        (map #(str/replace % #"^refs/tags/" ""))))
 
+(defn- sh-git-ls-remote
+  [url]
+  (loop []
+    (let [{:keys [exit err] :as res} (sh/sh "git" "ls-remote" url)]
+      (cond
+        (= 0 exit)
+        res
+
+        (and (< 0 exit) (not (str/includes? err "Operation timed out")))
```

Comment:
> "Operation timed out"

Messages text can be changed, so I'd like to avoid branching by message texts as much as possible.
If there is a specific exit code (do you know?), we should use that.
If not, we can leave it as it is.

Reply:
The exit code with this error message is 128. As is almost all git command exit codes :-/